code/datums/spells/shadow_walk.dm

```diff
@@ -0,0 +1,91 @@
+/obj/effect/proc_holder/spell/targeted/shadowwalk
+	name = "Shadow Walk"
+	desc = "Grants unlimited movement in darkness."
+	charge_max = 0
+	clothes_req = 0
+	selection_type = "range"
+	range = -1
+	include_user = TRUE
+	cooldown_min = 0
+	overlay = null
+	action_icon_state = "ninja_cloak"
+	action_background_icon_state = "bg_alien"
+
+/obj/effect/proc_holder/spell/targeted/shadowwalk/cast(list/targets, mob/user = usr)
+	var/L = user.loc
+	if(istype(user.loc, /obj/effect/dummy/shadow))
+		var/obj/effect/dummy/shadow/S = L
+		S.end_jaunt(FALSE)
+		return
+	else
+		var/turf/T = get_turf(user)
+		var/light_amount = T.get_lumcount()
+		if(light_amount < SHADOW_SPECIES_LIGHT_THRESHOLD)
+			playsound(get_turf(user), 'sound/magic/ethereal_enter.ogg', 50, 1, -1)
+			visible_message("<span class='boldwarning'>[user] melts into the shadows!</span>")
+			var/obj/effect/dummy/shadow/S2 = new(get_turf(user.loc))
+			user.forceMove(S2)
```

Comment:
Is the user able to drop items during that time? If so, they may be lost when the jaunt ends.